presto-main/src/main/java/io/prestosql/sql/planner/planprinter/IoPlanPrinter.java

```diff
@@ -600,19 +616,90 @@ protected Void visitPlan(PlanNode node, IoPlanBuilder context)
         }
 
         @Override
-        public Void visitTableScan(TableScanNode node, IoPlanBuilder context)
+        public Void visitFilter(FilterNode node, IoPlanBuilder context)
+        {
+            return visitScanFilterAndProjectInfo(node, Optional.of(node), Optional.empty(), context);
+        }
+
+        @Override
+        public Void visitProject(ProjectNode node, IoPlanBuilder context)
+        {
+            if (node.getSource() instanceof FilterNode) {
+                return visitScanFilterAndProjectInfo(node, Optional.of((FilterNode) node.getSource()), Optional.of(node), context);
+            }
+
+            return visitScanFilterAndProjectInfo(node, Optional.empty(), Optional.of(node), context);
+        }
+
+        private Void visitScanFilterAndProjectInfo(
+                PlanNode node,
+                Optional<FilterNode> filterNode,
+                Optional<ProjectNode> projectNode,
+                IoPlanBuilder context)
+        {
+            checkState(projectNode.isPresent() || filterNode.isPresent());
+
+            PlanNode sourceNode;
+            if (filterNode.isPresent()) {
+                sourceNode = filterNode.get().getSource();
+            }
+            else {
+                sourceNode = projectNode.get().getSource();
+            }
+
+            Optional<TableScanNode> scanNode;
+            if (sourceNode instanceof TableScanNode) {
+                scanNode = Optional.of((TableScanNode) sourceNode);
+            }
+            else {
+                scanNode = Optional.empty();
+            }
+
+            if (scanNode.isPresent() && filterNode.isPresent()) {
+                addConstraintsFromFilter(filterNode.get(), scanNode.get(), context);
+                return null;
+            }
+
+            sourceNode.accept(this, context);
+            return null;
+        }
+
+        private void addConstraintsFromFilter(FilterNode filterNode, TableScanNode tableScanNode, IoPlanBuilder context)
```

Comment:
Instead of passing the `FilterNode` here can we pass the  filter `Expression` ? So when we visit a normal TableScan we compute the column details like 
```
addConstraintsFromFilter(TRUE_LITERAL, tableScanNode, context)
```

When visiting a FilterNode we can check for its source and compute i/o based on its source